DOCS-6510 Changes (#16)
mulesoft/docs-functional-monitoring: modules/ROOT/pages/bat-example-test-suite.adoc

  # - type: PagerDuty
  #   options:
  #     PAGER_DUTY_ROUTING_KEY: <routing_key>
  # - type: Slack
  #   options:
  #     SLACK_WEBHOOK: <webhook_url>
  # - type: Email
  #   options:
  #     EMAILS: <email_list>
  # - type: NewRelic
  #   options:
  #     NEW_RELIC_LICENSE_KEY: <license_key>
  # - type: JSON
  #   outFile: /tmp/JSON.json
  # - type: JUnit
  #   outFile: /tmp/JUnit.xml
----

Every test suite must contain a `bat.yaml` file at its root level.

=== The `suite` Section of the example `bat.yaml` File
The `suite` section names the suite. You can use any names that you like for the suites that you write.

=== The `files` Section of the example `bat.yaml` File

This example test suite contains only one test. The test is listed in the `files` section as the file `HelloWorld.dwl` in the `tests` folder.

As you create tests, you add references to them in your `bat.yaml` file. For example, suppose that in your test-suite directory you create a folder named `api-tests` and write a test named `first-test.dwl` in that folder. In the `files` section of your `bat.yaml` file, you add the following indented line:

----
  file: api-tests/first-test.dwl
----

> If you write a second test, this one named `second-test.dwl`, you add this line after that one:

Reviewer: Can we avoid "this line after that one" and still retain intent?
```suggestion
If you write a second test, this one named `second-test.dwl`, you add this line next:
```

Author: Done.

----
files:
  file: retail-site-tests/orderFullfillment.dwl
  file: retail-site-tests/shoppingCart.dwl
----

To run the `orderFullfillment.dwl` test after the `shoppingCart.dwl` test, you can change the order in which they are listed:

----
files:
  file: retail-site-tests/shoppingCart.dwl
  file: retail-site-tests/orderFullfillment.dwl
----

Finally, you can create more than one folder to contain the tests in a test suite. You can even create subfolders within the folders that contain your tests. Just be sure that you provide the correct relative path when you list a test in the `files` section of your `bat.yaml` file.

=== The `reporters` Section of the `bat.yaml` File

This section is for configuring various tools or third-party applications to report test failures. In this example, a failure of the `HelloWorld.dwl` test prompts the BAT CLI to write a report of the failure to the file `/tmp/HTML.html`. For information about setting up reports, see xref:bat-reporting-task.adoc[Configure and View Test Reports].

> === The `secrets` Section of the `bat.yaml` file

Reviewer: file > File
```suggestion
=== The `secrets` Section of the `bat.yaml` File
```

Author: Done.

> If you plan to run a test suite from a private location and the APIs that you plan to test are deployed on CloudHub, you can use the BAT CLI's integration with Anypoint Secrets Manager to use shared secrets in your tests.

Reviewer 2: the APIS can be deployed anywhere. The Location must be private.
The integration is for all AFM not the CLI only. Also you can use secrets in the bat.yaml for reporters for example.

Author: Resolved.

=== The `config` Folder

> You can store variables in the files in the `config` folder. For example, let's say you're developing an API and you deploy it to two different environments: `dev` and `qa`. The endpoint url for the first is `dev.myapi.com` and for the second `qa.myapi.com`. The API should behave the same in both environments.

Reviewer: replacing "let's say" to ease translation
```suggestion
You can store variables in the files in the `config` folder. For example, imagine you're developing an API and you deploy it to two different environments: `dev` and `qa`. The endpoint URL for the first is `dev.myapi.com` and for the second `qa.myapi.com`. The API should behave the same in both environments.
```

Reviewer: Is there a disconnect in saying that the API should behave the same in both environments, followed by saying that you want to test it in each?
Also, is "should" a direction or a prediction?

Reviewer: GLOBAL: I recommend spelling out "configuration," when not referring to a syntax value or variable.
I haven't marked it, but "config" appears twice in the following text, and most likely elsewhere.
Suggest search and replace if you agree.

Author: Done.

.The content of `dev.dwl`, the config file for the `dev` environment
----
{
url: "dev.myapi.com"
}
----

.The content of `qa.dwl`, the config file for the `qa` environment
----
{
url: "qa.myapi.com"
}
----

In your test, you can use the `url` variable by placing it within this syntax: `$(config.variable)`, where _variable_ is the name of the variable that you want to use. Here is an example test that uses the `url` variable in a `GET` request to test the assertion that the response code must be 200:
[linenums]
----
bat::BDD
import * from bat::Assertions
---
suite("Hello world suite") in [
  it must 'answer 200' in [
    GET `$(config.url)` with {} assert [
      $.response.status mustEqual 200
  ]
]
----

When you run the test, you can use the `--config` parameter to specify which config file to use. (If you upload your test suite as a monitor to API Functional Monitoring in Anypoint Platform, you can specify the configuration to use when you create a schedule for the monitor.)

Use JSON for the format of your config files, and `.dwl` as the file extension. You can put as many variables as you need in each file.

=== The `tests` folder

> In the example test suite, the test -- `HelloWorld.dwl` -- is in a folder named `tests`. As explained in the section above about the `bat.yaml` file in this example, the `files` section in the `bat.yaml` file points to the tests in a test suite.

Reviewer: GLOBALLY replace dashes (and their keyboard replacements using hyphens and spaces).
In this text, you don't need anything.
Also, GLOBALLY remove references to content "above" or "below."
```suggestion
In the example test suite, the test `HelloWorld.dwl` is in a folder named `tests`. As explained in the section  about the `bat.yaml` file in this example, the `files` section in the `bat.yaml` file points to the tests in a test suite.
```

Author: Done.
